elastic/kibana
# [Event Log][8.0] Set all event log index assets to hidden

Resolves https://github.com/elastic/kibana/issues/58035

## Summary

* Updates event log index creation logic so that
  * Newly created index templates are no longer legacy templates. Existing index templates will still be legacy.
  * Newly created index templates, indices and index aliases are set to hidden
* Checks for existing event log templates, indices and aliases and sets them to hidden if they are not already hidden.

## To Verify
* Ideally, you want to use data from a previous version like 7.15 and start up this branch with it.
* Verify that previous and new event log index templates have `index.hidden: true`
  * `https://localhost:9200/_template/.kibana-event-log*` for 7.15 event log index (legacy endpoint)
  * `https://localhost:9200/_index_template/.kibana-event-log*` for 8.0 event log index (new endpoint)
* Verify that previous and new event log index settings have `index.hidden: true`
  * `https://localhost:9200/.kibana-event-log*/_settings`
* Verify that previous and new event log index aliases have `is_hidden: true`
  * `https://localhost:9200/_alias/.kibana-event-log*`
* Verify that no event log indices are listed when doing CAT indices: `https://localhost:9200/_cat/indices`
* Go to the index template UI and see that the kibana-event-log 8.0 index template is not listed under legacy index templates.
* Verify that you can still create a rule with actions and have event log documents written.

### Checklist

Delete any items that are not applicable to this PR.
